app/services/v1x0/catalog/add_to_order.rb

```diff
@@ -0,0 +1,30 @@
+module V1x0
```

Comment:
@syncrou Could this module name spacing match the one that's being used in the controllers
module Api
  module V1

And then let the scoping pick up the appropriate service instance from the controller then we won't have to do
icon = V1x0::Catalog::CreateIcon.new(icon_params).process.icon
it could be
icon = Service::Catalog::CreateIcon.new(icon_params).process.icon

With no version string in the name.

Reply:
@eclarizio and I worked at attempting to make the above scenario work yesterday and we were not able to get the class resolution to work in the simple way suggested above.

We also went so far as to add in an `api` namespace to duplicate the namespacing in the controllers file tree:  e.g. `app/services/api/v1x0` thinking that was our issue, but again ruby would complain about the constant not being found.

I definitely want to figure out a solution to this problem but so far we haven't.

Reply:
@syncrou could it be because of the auto loading of constants or the config.auto_load_paths, we could force load the services directory in the application.rb and see if that resolves the constant issue.

Reply:
@mkanoor I tried that yesterday, rails autoloads anything under the `app` directory, but yeah I tried to manually add `app/services` and it didn't work :(